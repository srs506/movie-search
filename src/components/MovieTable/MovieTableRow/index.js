import * as React from 'react';
import { Star } from 'react-feather';

const MovieTableRow = ({ movie }) => {
  return (
    <tr>
      <td>
        <img
          src={`${movie.basePosterPath}${movie.poster_path}`}
          alt={`Poster for '${movie.title}'`}
          width="92px;"
          className="posterImage"
        />
      </td>
      <td>{movie.title}</td>
      <td>
        <div className="voteAverage">
          <Star className="voteAverageIcon" />
          {movie.vote_average}
        </div>
      </td>
      <td>{new Date(movie.release_date).getFullYear()}</td>
    </tr>
  );
};

export default MovieTableRow;
