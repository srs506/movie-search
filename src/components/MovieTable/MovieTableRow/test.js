import React from 'react';
import ReactDOM from 'react-dom';
import MovieTableRow from './index';
import renderer from 'react-test-renderer';

const movie = {
  vote_count: 4804,
  id: 346364,
  video: false,
  vote_average: 7.2,
  title: 'It',
  popularity: 851.149606,
  poster_path: '/9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg',
  original_language: 'en',
  original_title: 'It',
  genre_ids: [18, 14, 27, 53],
  backdrop_path: '/tcheoA2nPATCm2vvXw2hVQoaEFD.jpg',
  adult: false,
  overview:
    'In a small town in Maine, seven children known as The Losers Club come face to face with life problems, bullies and a monster that takes the shape of a clown called Pennywise.',
  release_date: '2017-09-05',
  basePosterPath: 'https://image.tmdb.org/t/p/w154'
};

it('renders without crashing', () => {
  const tableBody = document.createElement('tbody');
  ReactDOM.render(<MovieTableRow movie={movie} />, tableBody);
});
