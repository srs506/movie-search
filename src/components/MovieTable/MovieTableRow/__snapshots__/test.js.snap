// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has a valid snapshot 1`] = `
<tr>
  <td>
    <img
      alt="Poster for 'It'"
      className="posterImage"
      src="https://image.tmdb.org/t/p/w154/9E2y5Q7WlCVNEhP5GiVTjhEhx1o.jpg"
      width="92px;"
    />
  </td>
  <td>
    It
  </td>
  <td>
    <div
      className="voteAverage"
    >
      <svg
        className="voteAverageIcon"
        fill="none"
        height="24"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polygon
          points="12 2 15.09 8.26 22 9.27 17 14.14 18.18 21.02 12 17.77 5.82 21.02 7 14.14 2 9.27 8.91 8.26 12 2"
        />
      </svg>
      7.2
    </div>
  </td>
  <td>
    2017
  </td>
</tr>
`;
